include/secure_services.h

```diff
@@ -66,6 +66,17 @@ int spm_request_random_number(u8_t *output, size_t len, size_t *olen);
  */
 int spm_request_read(void *destination, u32_t addr, size_t len);
 
+/** Check if S0 is the active B1 slot.
+ *
+ * @param[in]   s0_address Address of s0 slot.
+ * @param[in]   s1_address Address of s1 slot.
+ * @param[out]  s0_active  Set to 'true' if s0 is active slot, 'false' otherwise
+ *
+ * @retval 0        If successful.
+ * @retval -EINVAL  If info for both slots are NULL.
+ */
+int spm_s0_active(u32_t s0_address, u32_t s1_address, bool *s0_active);
```

Comment:
I personally think this should return the active slot instead of the bool s0_active, especially since s0_active's value is the opposite of the active slot (s0 active -> 1, s1 active -> 0).
```suggestion
int spm_s0_active(u32_t s0_address, u32_t s1_address, u32_t *active_slot);
```
Alternatively, if it would follow convention for int retvals, you could return active_slot through the retval instead.

Reply:
I don't like the `u32` solution. Why do we need to be capable of representing more than two values? How should the caller act if its not 0 or 1?

IMO it desirable to follow the convention in this file, and also the error message provides useful information to the caller. If we were to say `return slot number, or negative error number if fail` then the calling code would not look as good.

```
int slot;

slot = spm_get_active_slot(PM_S0_ADDR, PM_S1_ADDR);

if (slot < 0 || slot > 1) {
   LOG_ERR("Failed to read slot")
}

if (slot == 0) { /* magic value, should define this 0 */
 /* bla */
}
```

compared to 
```
int err;
bool s0_active;

err = spm_s0_active(PM_S0_ADDRESS, PM_S1_ADDRESS, &s0_active);
if (err != 0) {
   LOG_ERR("Failed reading active slot")
}

if (s0_active) {
   /* bla */
}
```

Reply:
I would never use the value itself, so to me it made sense to have the typical case as the positive check (s0 would be active in most cases as bootloader updates are rare, was my intuition).

The only downside I see with your proposals is that you need to have defines for the magic values '0' and '1', which I would say complicates (marginally) the code.